app/api/views/mailbox.py

```diff
@@ -78,19 +78,47 @@ def delete_mailbox(mailbox_id):
     Delete mailbox
     Input:
         mailbox_id: in url
+        (optional) transfer_aliases_to: in body. Id of the new mailbox for the aliases.
+                                        If omitted or the value is set to -1,
+                                        the aliases of the mailbox will be deleted too.
     Output:
         200 if deleted successfully
 
     """
     user = g.user
-    mailbox = Mailbox.get(mailbox_id)
+    mailbox = Mailbox.get(id=mailbox_id)
 
     if not mailbox or mailbox.user_id != user.id:
         return jsonify(error="Forbidden"), 403
 
     if mailbox.id == user.default_mailbox_id:
         return jsonify(error="You cannot delete the default mailbox"), 400
 
+    data = request.get_json() or {}
+    if data.get("transfer_aliases_to") and (int(data.get("transfer_aliases_to")) >= 0):
+        new_mailbox_id = int(data.get("transfer_aliases_to"))
+        new_mailbox = Mailbox.get(id=new_mailbox_id)
+
+        if not new_mailbox or new_mailbox.user_id != user.id:
+            return jsonify(error="You must transfer the aliases to a mailbox you own."), 403
+
+        if new_mailbox_id == mailbox_id:
+            return jsonify(error="You can not transfer the aliases to the mailbox you want to delete."), 400
+
+        if not new_mailbox.verified:
+            return jsonify(error="Your new mailbox is not verified"), 400
+
+        for alias in mailbox.aliases:
```

Comment:
This can take a lot of time for a mailbox that has a lot of aliases. 
2 approaches can be used here:
1. Convert this to a async task. An example is https://github.com/simple-login/app/blob/6bf2a58643daf70f049b1afe8bea53ed6724ffdb/app/dashboard/views/mailbox.py#L87
2. Use a SQL bulk update. The challenge would be to update the `AliasMailbox` table but ignore aliases that are owned by both current and new mailboxes. For the Alias table, something like  `Alias.filter_by(mailbox_id=mailbox.id).update({Alias.mailbox_id: {cls.mailbox_id: to_mailbox_id})` should work.